code/modules/customitems/item_defines.dm

```diff
@@ -235,6 +235,29 @@
 		to_chat(target, "<span class='notice'>You comb your tail with the [src].</span>")
 		used = 1
 
+/obj/item/device/fluff/desolate_baton_kit //DesolateG: Michael steampunk_witch
+	name = "stun baton converstion kit"
+	desc = "Some sci-fi looking parts for a stun baton."
+	icon = 'icons/obj/custom_items.dmi'
+	icon_state = "scifikit"
+	w_class = 2
+
+/obj/item/device/fluff/desolate_baton_kit/afterattack(atom/target, mob/user, proximity)
+	if(!proximity || !ishuman(user) || user.lying)
+		return
+
+	if(istype(target, /obj/item/weapon/melee/baton) && !istype(target, /obj/item/weapon/melee/baton/cattleprod))
+		to_chat(user, "<span class='notice'>You modify the appearance of [target].</span>")
+		//because batons use the initial() proc, we can't just swap the icon state, sadly
+		var/obj/item/weapon/melee/baton/the_baton = target
```

Comment:
quick question; if you change the `icon` here too, will it still work; that way you can keep the baton icons in the custom items as opposed to polluting weapons.dmi.